integration/setup.go

```diff
@@ -50,110 +52,115 @@ var (
 )
 
 // SetupOpts represents setup options.
+//
+// TODO add option to use read-only user
 type SetupOpts struct {
+	// If true, returns two client connections to different systems for compatibility test.
+	CompatTest bool
+
 	// Database to use. If empty, temporary test-specific database is created.
 	DatabaseName string
 
 	// Data providers.
 	Providers []shareddata.Provider
 }
 
-// SetupWithOpts setups the test according to given options,
-// and returns test-specific context (that is canceled when the test ends), database collection
-// and the port of the running server.
-func SetupWithOpts(t *testing.T, opts *SetupOpts) (context.Context, *mongo.Collection, int) {
-	t.Helper()
+// SetupResult represents setup results.
+type SetupResult struct {
```

Comment:
Maybe something like this? In case we need to extend such a config later, that could simplify/speed up dev process.

```go
struct System {
	Collection       *mongo.Collection
	Port             uint16
}

type SetupResult struct {
	Ctx           context.Context
	TargetSystem        System
	ReferenceSystem  System
}
```

Reply:
I would apply YAGNI there: we have only one test that needs ports, and most tests use simple `Setup` or `SetupCompat` functions